Code/src/main/java/nl/utwente/group10/ui/components/anchors/ConnectionAnchor.java

```diff
@@ -4,48 +4,140 @@
 import java.util.List;
 import java.util.Optional;
 
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.value.ObservableValue;
+import javafx.fxml.FXML;
 import javafx.geometry.Point2D;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
 import javafx.scene.shape.Circle;
+import javafx.scene.shape.Shape;
+import nl.utwente.group10.haskell.exceptions.HaskellTypeError;
+import nl.utwente.group10.haskell.hindley.HindleyMilner;
 import nl.utwente.group10.haskell.type.Type;
+import nl.utwente.group10.ui.BackendUtils;
 import nl.utwente.group10.ui.CustomUIPane;
 import nl.utwente.group10.ui.components.ComponentLoader;
+import nl.utwente.group10.ui.components.ConnectionStateDependent;
 import nl.utwente.group10.ui.components.blocks.Block;
+import nl.utwente.group10.ui.components.blocks.output.OutputBlock;
 import nl.utwente.group10.ui.components.lines.Connection;
+import nl.utwente.group10.ui.handlers.ConnectionCreationManager;
 
 /**
  * Represents an anchor of a Block that can connect to (1 or more) Connections.
  * 
+ * Has an invisible part that acts as an enlargement of the touch zone.
+ * 
  * The primary Connection (if present) is the first element in getConnections().
  * This means that the oldest Connection is the primary connection.
+ * 
+ * The ConnectionAnchor keeps track of its accepted type (signature), and will typecheck this for new connections.
  */
-public abstract class ConnectionAnchor extends Circle implements ComponentLoader {
-    /** The pane on which this Anchor resides. */
-    private CustomUIPane pane;
+public abstract class ConnectionAnchor extends StackPane implements ComponentLoader, ConnectionStateDependent {
 
     /** The block this Anchor is connected to. */
     private Block block;
 
     /** The connections this anchor has, can be empty for no connections. */
     private List<Connection> connections;
+    
+    /** The visual representation of the ConnectionAnchor. */
+    @FXML private Shape visibleAnchor;
+    
+    /** The invisible part of the ConnectionAnchor (the touchZone). */
+    @FXML private Shape invisibleAnchor;
+    
+    /** The signature that is accepted by this anchor. */
+    private Type signature;
+    
+    /** Property storing the error state. */
+    private BooleanProperty isError;
+    
+    /** Property storing the active state. */
```

Comment:
What does 'active' mean here? Is it the same as TactilePane's active state, that is, 'this node is currently being dragged'?